src/seedu/addressbook/AddressBook.java

```diff
@@ -554,11 +561,43 @@ private static boolean isDisplayIndexValidForLastPersonListingView(int index) {
      * @param deletedPerson successfully deleted
      * @return successful delete person feedback message
      */
-    private static String getMessageForSuccessfulDelete(String[] deletedPerson) {
+    private static String getMessageForSuccessfulDelete(HashMap<PersonProperty,String> deletedPerson) {
         return String.format(MESSAGE_DELETE_PERSON_SUCCESS, getMessageForFormattedPersonData(deletedPerson));
     }
 
     /**
+     * Sort and display list of all persons in alphabetical order
+     * 
+     * @return display feedback message for the operation's result
+     */
+    private static String executeSortAllPersonsInAddressBook() {
+    	ArrayList<HashMap<PersonProperty,String>> personsToSort = getAllPersonsInAddressBook();
+    	ArrayList<HashMap<PersonProperty,String>> toBeDisplayed = sortPersonsInAlphabeticalOrder(personsToSort);
+        showPersonToUser(toBeDisplayed);
+        return getMessageForPersonsDisplayedSummary(toBeDisplayed);
+    }
+    
+    /**
+     * Sort incoming list of persons in alphabetical order by name
+     * 
+     * @param list of persons to be sorted
+     * @return list of sorted persons
+     */
+    private static ArrayList<HashMap<PersonProperty,String>> sortPersonsInAlphabeticalOrder(ArrayList<HashMap<PersonProperty,String>> personsToSort) {
```

Comment:
this is a particularly long line of code; please check the coding standards to see how best you can split this line.